cmd/lncli/psbt.go

```diff
@@ -0,0 +1,311 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/signal"
+	"github.com/urfave/cli"
+)
+
+const (
+	userMsgFund = `PSBT funding initiated with peer %x.
+Please create a PSBT that sends %v (%d satoshi) to the funding address %s.
+
+Example with bitcoind:
+	bitcoin-cli walletcreatefundedpsbt [] '[{"%s":%.8f}]'
+
+Or if you are using a wallet that can fund a PSBT directly (currently not
+possible with bitcoind), you can use this PSBT that contains the same address
+and amount: %s
+
+Paste the funded PSBT here to continue the funding flow.
+Base64 encoded PSBT: `
+
+	userMsgSign = `
+PSBT verified by lnd, please continue the funding flow by signing the PSBT by 
+all required parties/devices. Once the transaction is fully signed, paste it
+again here.
+
+Base64 encoded PSBT: `
+)
+
+// openChannelPSBT starts an interactive channel open protocol that uses a
+// partially signed bitcoin transaction (PSBT) to fund the channel output. The
+// protocol involves several steps between the RPC server and the CLI client:
+//
+// RPC server                           CLI client
+//     |                                    |
+//     |  |<------open channel (stream)-----|
+//     |  |-------ready for funding----->|  |
+//     |  |<------PSBT verify------------|  |
+//     |  |-------ready for signing----->|  |
+//     |  |<------PSBT finalize----------|  |
+//     |  |-------channel pending------->|  |
+//     |  |-------channel open------------->|
+//     |                                    |
+func openChannelPSBT(ctx *cli.Context, client lnrpc.LightningClient,
+	req *lnrpc.OpenChannelRequest, basePsbt string, block bool) error {
```

Comment:
I wanted to avoid the `psbt.go` file containing hard coded strings of flag names as they would likely be forgotten there if we ever changed them in the main file.